agreement_contract/models/agreement.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class Agreement(models.Model):
+    _inherit = 'agreement'
+
+    contract_type = fields.Selection(
+        selection=[
+            ('sale', 'Customer'),
+            ('purchase', 'Supplier'),
+        ],
+        default="sale",
+    )
+    contract_count = fields.Integer(
+        string="Contract_count",
+        compute="_compute_contract_count",
+    )
+
+    @api.multi
+    def _compute_contract_count(self):
+        for contract in self:
+            search_contract = contract.env['account.analytic.account'].\
+                search([('agreement_id', '=', self.id)])
+            if search_contract:
+                contract.contract_count = len(search_contract)
+
+    @api.multi
+    def action_view_contract(self):
+        contract = self.env['account.analytic.account']
+        search_contract = contract.search([('agreement_id', '=', self.id)])
+        if not search_contract:
+            raise UserError(
+                _('Please create contract.'))
```

Comment:
I think should to add contract_types after this line
contract_types = list(set(self.mapped('contract_type')))
if len(contract_types) > 1:
raise UserError(_('Please select same contract type.'))